aggregators/rules/ipg.js

```diff
@@ -1,17 +1,49 @@
+const jsDOM = require("jsdom");
 class IPG {
     constructor() {
-        this.Location = "http://blogs.magicjudges.org/rules/ipg/";
+        this.location = "http://blogs.magicjudges.org/rules/ipg";
     }
 
     find(parameter, callback) {
-        //todo
+        jsDOM.env(
+            this.location + parameter,
+            ["http://code.jquery.com/jquery.js"],
+            /**
+             *
+             * @param err
+             * @param window
+             * @param window.$ jquery
+             * @param window.$.each jquery for each loop
+             * @param window.$.prop jquery property
+             */
+            function(err,window){
+                let resultLength = (window.$(".entry-header h1:first").text()).length+6;
+                const result = ["**"+window.$(".entry-header h1:first").text()+"**\n"];
+                window.$(".alert-info").remove();
+                window.$(".entry-content h2, .entry-content p, .entry-content li, .entry-content td").each(function(){
+                    let text = window.$(this).text()+"\n";
+                    if(window.$(this).prop("nodeName").toLocaleLowerCase()==="h2"){
+                        text = "\n**"+text+"**";
+                    }
+                    if(resultLength+text.length>2000){
```

Comment:
I will change it to a constant because the jquery.each() function can't reach class properties.